File: medium/3876.py
"""
3876. Construct Uniform Parity Array II

https://leetcode.com/problems/construct-uniform-parity-array-ii/

Weekly Contest 494
"""


class Solution:
    def uniformArray(self, nums1: list[int]) -> bool:
        min_odd: int | None = None
        min_even: int | None = None

        for num in nums1:
            if num & 1:
                if min_odd is None or num < min_odd:
                    min_odd = num
            else:
                if min_even is None or num < min_even:
                    min_even = num

        return not min_odd or not min_even or min_even > min_odd

File: medium/3876.rs
// 3876. Construct Uniform Parity Array II
// https://leetcode.com/problems/construct-uniform-parity-array-ii/
// Weekly Contest 494

impl Solution {
    pub fn uniform_array(nums1: Vec<i32>) -> bool {
        let mut min_odd: i32 = std::i32::MAX;
        let mut min_even: i32 = std::i32::MAX;

        for num in nums1 {
            if num & 1 == 0 {
                if min_even > num {
                    min_even = num;
                }
            } else {
                if min_odd > num {
                    min_odd = num;
                }
            }
        }

        (min_odd == std::i32::MAX) || (min_even == std::i32::MAX) || (min_even > min_odd)
    }
}
